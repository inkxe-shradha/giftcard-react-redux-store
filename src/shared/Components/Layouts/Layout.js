import React from "react";
import Footer from "./footer/Footer";
import Header from "./header/Header";
import { makeStyles } from "@mui/styles";
import { useLocation } from "react-router-dom";
import { toast, ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { logOutUser } from "../../../store/actions/authAction";
import { useGoogleLogout } from "react-google-login";
import Spinner from "../Spinner/Spinner";
const useStyles = makeStyles((theme) => ({
  page: {
    background: theme.palette.background.default,
    width: "100%",
    flexGrow: 1,
    minHeight: "100vh",
    paddingBlock: theme.spacing(1),
  },
  toolbar: theme.mixins.toolbar,
}));
const Layout = ({ children, user, isAuthenticated, logout, loading }) => {
  const { signOut, loaded } = useGoogleLogout({
    clientId: process.env.REACT_APP_GOOGLE_CLIENT_ID,
    onLogoutSuccess,
  });

  function onLogoutSuccess() {
    console.log("Logged out successfully");
  }
  const { pathname } = useLocation();
  const classes = useStyles();
  const handelLoggingOut = () => {
    signOut();
    logout();
    toast.success("Logged out successfully");
  };
  return (
    <>
      {pathname !== "/login" && (
        <Header
          user={user}
          authStatus={isAuthenticated}
          logOut={handelLoggingOut}
        />
      )}
      {/* Main content */}
      <div className={classes.page}>
        {pathname !== "/login" && <div className={classes.toolbar}></div>}
        {children}
      </div>
      <Footer />
      <ToastContainer
        autoClose={3000}
        hideProgressBar
        position="top-right"
        newestOnTop={true}
        closeOnClick
        rtl={false}
        pauseOnHover
        theme="dark"
      />
      <Spinner isLoading={loading} />
    </>
  );
};

const mapStateToProps = (state) => ({
  loading: state.loading.loading,
  user: state.auth.user,
  isAuthenticated: state.auth.isAuthenticated,
});

const mapDispatchToProps = {
  logout: logOutUser,
};
// Prototypes
Layout.propTypes = {
  loading: PropTypes.bool.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  user: PropTypes.object,
};

export default connect(mapStateToProps, mapDispatchToProps)(Layout);
