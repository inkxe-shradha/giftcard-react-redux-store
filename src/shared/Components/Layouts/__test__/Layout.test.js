import { createMemoryHistory } from "history";
import { mount } from "enzyme";
import { ThemeProvider } from "@mui/system";
import { BrowserRouter as Router } from "react-router-dom";
import { createTheme } from "@mui/material";
import Layout from "../Layout";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Layout test suite", () => {
  const theme = createTheme();
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const state = {
    auth: {
      user: {},
      isAuthenticated: false,
    },
    loading: {
      loading: false,
    },
  };
  const mountRender = (component, location, initialState = state) => {
    const history = createMemoryHistory();
    history.push(location);
    const store = mockStore(initialState);
    return mount(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router location={history.location} navigator={history}>
            {component}
          </Router>
        </ThemeProvider>
      </Provider>
    );
  };
  it("Should create a snapshot of the layout  page", () => {
    const view = mountRender(<Layout />, "/");
    expect(view.html()).toMatchSnapshot();
  });
});
