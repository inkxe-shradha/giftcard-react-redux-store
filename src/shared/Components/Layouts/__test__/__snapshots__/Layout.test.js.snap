// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout test suite Should create a snapshot of the layout  page 1`] = `"<div class=\\"makeStyles-root-3\\"><header class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed makeStyles-appBar-6 mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root\\"><div class=\\"MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium makeStyles-menuButton-4 makeStyles-hiddenBigScreen-18 css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"menu\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"MenuIcon\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><h6 class=\\"MuiTypography-root MuiTypography-h6 makeStyles-title-5 css-2ulfj5-MuiTypography-root\\">YoYo Gifting</h6><div class=\\"makeStyles-hiddenSmallScreen-17 d-none\\"><a class=\\"MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" href=\\"/\\"> Home <span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></a><a class=\\"MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-1y942vo-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" href=\\"/gift-received\\"> Gift Received <span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></a><a class=\\"MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-1y942vo-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" href=\\"/gift-sent\\"> Gift Sent <span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></a></div><div class=\\"makeStyles-signedInUser-16\\"><p class=\\"MuiTypography-root MuiTypography-body1 css-18brw8f-MuiTypography-root\\">Hello, Guest!</p></div><div class=\\"makeStyles-signedInUser-16 makeStyles-hideMenu-19\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"AccountCircleIcon\\" aria-haspopup=\\"true\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z\\"></path></svg></div><div><a class=\\"MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-1y942vo-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" href=\\"/login\\">Login<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></a></div></div></header><nav class=\\"makeStyles-navDrawer-10\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-drawerPaper-9 css-joixkg-MuiPaper-root\\"></div></nav></div><div class=\\"makeStyles-page-1\\"><div class=\\"makeStyles-toolbar-2\\"></div></div><footer class=\\"mt-3\\"><p class=\\"MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root\\">Made with ❤️ by Shrdha. Copyright © <a color=\\"inherit\\" href=\\"/\\">Your Website</a> 2022.</p></footer><div class=\\"Toastify\\"></div><div><div aria-hidden=\\"true\\" class=\\"MuiBackdrop-root css-1ji7omv-MuiBackdrop-root\\" style=\\"opacity: 0; visibility: hidden;\\"><span class=\\"MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root\\" style=\\"width: 40px; height: 40px;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></span></div></div>"`;
